src/main/java/com/powsybl/openloadflow/ac/equations/AbstractClosedBranchAcFlowEquationTerm.java

```diff
@@ -30,24 +29,33 @@
 
     protected final Variable ph2Var;
 
-    protected final List<Variable> variables;
+    protected Variable a1Var;
 
-    protected final double a1;
+    protected Variable a2Var;
 
-    protected final double a2;
+    protected final List<Variable> variables;
 
-    protected AbstractClosedBranchAcFlowEquationTerm(LfBranch branch, LfBus bus1, LfBus bus2, VariableSet variableSet) {
+    protected AbstractClosedBranchAcFlowEquationTerm(LfBranch branch, LfBus bus1, LfBus bus2, VariableSet variableSet,
+                                                     boolean deriveA1, boolean deriveA2) {
         super(branch);
         Objects.requireNonNull(bus1);
         Objects.requireNonNull(bus2);
+        Objects.requireNonNull(variableSet);
         v1Var = variableSet.getVariable(bus1.getNum(), VariableType.BUS_V);
         v2Var = variableSet.getVariable(bus2.getNum(), VariableType.BUS_V);
         ph1Var = variableSet.getVariable(bus1.getNum(), VariableType.BUS_PHI);
         ph2Var = variableSet.getVariable(bus2.getNum(), VariableType.BUS_PHI);
-        variables = ImmutableList.of(v1Var, v2Var, ph1Var, ph2Var);
-        PiModel piModel = branch.getPiModel();
-        a1 = piModel.getA1();
-        a2 = piModel.getA2();
+        ImmutableList.Builder<Variable> variablesBuilder = ImmutableList.<Variable>builder()
+                .add(v1Var, v2Var, ph1Var, ph2Var);
+        if (deriveA1) {
+            a1Var = variableSet.getVariable(branch.getNum(), VariableType.BRANCH_ALPHA1);
+            variablesBuilder.add(a1Var);
+        }
+        if (deriveA2) {
```

Comment:
In a way our code is not so clear: dot we allow or not to have a PTC at side 2 ? The question will be raised for RTC too ? I think that if we Don't allow it from the Pi Model, we have to be consistent for the rest of the code. What do you think ?

Reply:
There is no PTC on side 2 in IIDM. I kept most of the code for side2 in the very hypothetical case of OLF would be integrated with another data model. 
But maybe you are right we should clean a2 and r2 everywhere in the code.